NullPointerException at PocketPlaneData.generatePocketPlane (Thaumic Horizons) crashing server (save corrupt in sp)

### Your GTNH Discord Username

@Роберт-_P0ZA_

### Your Pack Version

Daily 440

### Your Server

privat server

### Java Version

Java 25

### Type of Server

Vanilla Forge

### Your Actions

The server crashes immediately when a Primordial Pearl is thrown into a Planar Vortex to create a Pocket Plane. After the crash, the server enters a "crash loop" because it attempts to tick the erroring TileVortex upon startup. (its first planar on server)
Steps to Reproduce:
Setup a Planar Vortex (Thaumic Horizons).
Throw a Primordial Pearl into it to initiate Pocket Plane generation.
Server crashes during the generation process.

### Crash Report

https://pastebin.com/EsZFkV1Z

https://pastebin.com/BahaePeP  (i also try it in single world, same issue)

The crash is caused by a NullPointerException in the dimension generation thread.
java.lang.NullPointerException: Cannot assign field "cheat" because "vortex" is null
at com.kentington.thaumichorizons.common.lib.PocketPlaneData.generatePocketPlane(PocketPlaneData.java:115)

It appears that the vortex object is not being correctly passed to the PocketPlaneThread, or it's being GC'ed/invalidated before the thread starts, which is common in high-concurrency environments or newer Java versions.

Caused by: java.lang.NullPointerException: Cannot assign field "cheat" because "vortex" is null
        at Launch//com.kentington.thaumichorizons.common.lib.PocketPlaneData.generatePocketPlane(PocketPlaneData.java:115) ~[PocketPlaneData.class:?]
        at Launch//com.kentington.thaumichorizons.common.lib.PocketPlaneThread.run(PocketPlaneThread.java:34) ~[PocketPlaneThread.class:?]
        at java.base/java.lang.Thread.run(Thread.java:1474) ~[?:?]
        at Launch//com.kentington.thaumichorizons.common.tiles.TileVortex.createDimension(TileVortex.java:324) ~[TileVortex.class:?]

### Final Checklist

- [x] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [x] I can reproduce this crash consistently by following the exact steps I described above.
- [x] I have asked other people and they confirm they also crash by following the exact steps I described above.

--- a/src/main/java/com/kentington/thaumichorizons/common/lib/PocketPlaneData.java
+++ b/src/main/java/com/kentington/thaumichorizons/common/lib/PocketPlaneData.java
@@ -16,7 +16,6 @@ import java.util.Map;
 import java.util.Set;
 
 import net.minecraft.block.Block;
-import net.minecraft.tileentity.TileEntity;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.monster.EntitySlime;
 import net.minecraft.entity.passive.EntityChicken;
@@ -33,6 +32,7 @@ import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.potion.Potion;
 import net.minecraft.server.MinecraftServer;
+import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeGenBase;
@@ -79,7 +79,7 @@ public class PocketPlaneData {
     }
 
     public static void generatePocketPlane(final AspectList aspects, final PocketPlaneData data, final World world,
-                                           final int vortexX, final int vortexY, final int vortexZ, final int returnID) {
+            final int vortexX, final int vortexY, final int vortexZ, final int returnID) {
         if (!world.isRemote) {
             final int xCenter = 0;
             final int yCenter = 128;

--- a/src/main/java/com/kentington/thaumichorizons/common/lib/PocketPlaneData.java
+++ b/src/main/java/com/kentington/thaumichorizons/common/lib/PocketPlaneData.java
@@ -16,6 +16,7 @@ import java.util.Map;
 import java.util.Set;
 
 import net.minecraft.block.Block;
+import net.minecraft.tileentity.TileEntity;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.monster.EntitySlime;
 import net.minecraft.entity.passive.EntityChicken;
@@ -78,7 +79,7 @@ public class PocketPlaneData {
     }
 
     public static void generatePocketPlane(final AspectList aspects, final PocketPlaneData data, final World world,
-            final int vortexX, final int vortexY, final int vortexZ, final int returnID) {
+                                           final int vortexX, final int vortexY, final int vortexZ, final int returnID) {
         if (!world.isRemote) {
             final int xCenter = 0;
             final int yCenter = 128;
@@ -111,7 +112,11 @@ public class PocketPlaneData {
                 }
             }
             world.setBlock(xCenter, yCenter + 1, zCenter, ThaumicHorizons.blockVortex);
-            final TileVortex vortex = (TileVortex) world.getTileEntity(xCenter, yCenter + 1, zCenter);
+            TileEntity te = world.getTileEntity(xCenter, yCenter + 1, zCenter);
+            if (!(te instanceof TileVortex)) {
+                te = new TileVortex();
+            }
+            final TileVortex vortex = (TileVortex) te;
             vortex.cheat = true;
             vortex.returnID = returnID;
             vortex.dimensionID = PocketPlaneData.planes.size();

